### PR TITLE
Align PT and Flax API - allow loading checkpoint from PyTorch configs

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: tests/test_pipelines_flax.py [hunk 1]
```diff
@@ -57,6 +58,103 @@ def test_dummy_all_tpus(self):
...
+
+        images_pil = pipeline.numpy_to_pil(np.asarray(images.reshape((num_samples,) + images.shape[-3:])))
+        for i, image in enumerate(images_pil):
+            image.save(f"/home/patrick/images/flax-test-{i}_fp32.png")
+
+        assert images.shape == (8, 1, 512, 512, 3)
+        assert np.abs((np.abs(images[0, 0, :2, :2, -2:], dtype=np.float32).sum() - 0.05652401)) < 1e-3
+        assert np.abs((np.abs(images, dtype=np.float32).sum() - 2383808.2)) < 1e-2
+
+    def test_stable_diffusion_v1_4_bfloat_16(self):
```
Comment (patrickvonplaten) [inline tests/test_pipelines_flax.py:104]: no safety + outer pmap

File: tests/test_pipelines_flax.py [hunk 2]
```diff
@@ -57,6 +58,103 @@ def test_dummy_all_tpus(self):
...
+        prng_seed = jax.random.split(prng_seed, 8)
+        prompt_ids = shard(prompt_ids)
+
+        images = p_sample(prompt_ids, params, prng_seed, num_inference_steps).images
+
+        assert images.shape == (8, 1, 512, 512, 3)
+        assert np.abs((np.abs(images[0, 0, :2, :2, -2:], dtype=np.float32).sum() - 0.06652832)) < 1e-3
+        assert np.abs((np.abs(images, dtype=np.float32).sum() - 2384849.8)) < 1e-2
+
+    def test_stable_diffusion_v1_4_bfloat_16_with_safety(self):
```
Comment (patrickvonplaten) [inline tests/test_pipelines_flax.py:134]: with safety + inner pmap

File: tests/test_pipelines_flax.py [hunk 3]
```diff
@@ -57,6 +58,103 @@ def test_dummy_all_tpus(self):
...
+        prng_seed = jax.random.PRNGKey(0)
+        num_inference_steps = 50
+
+        num_samples = jax.device_count()
+        prompt = num_samples * [prompt]
+        prompt_ids = pipeline.prepare_inputs(prompt)
+
+        # shard inputs and rng
+        params = replicate(params)
+        prng_seed = jax.random.split(prng_seed, 8)
```
Comment (apolinario) [inline tests/test_pipelines_flax.py:153]: Better to replace `8` with `jax.device_count()`, no?

Comment (pcuenca) [reply tests/test_pipelines_flax.py:153]: 👍 agree

Comment (patrickvonplaten) [reply tests/test_pipelines_flax.py:153]: Yes this would indeed be a good idea (in case someone wants to open a PR for it, please feel free)

**Pull request merged**